src/Microsoft.CodeQuality.Analyzers/Core/Security/MarkVerbHandlersWithValidateAntiforgeryTokenAnalyzer.cs

```diff
@@ -0,0 +1,360 @@
+                this.HttpPatchAttributeSymbol = WellKnownTypes.HttpPatchAttribute(compilation);
+                this.AcceptVerbsAttributeSymbol = WellKnownTypes.AcceptVerbsAttribute(compilation);
+                this.NonActionAttributeSymbol = WellKnownTypes.NonActionAttribute(compilation);
+                this.ChildActionOnlyAttributeSymbol = WellKnownTypes.ChildActionOnlyAttribute(compilation);
+                this.HttpVerbsSymbol = WellKnownTypes.HttpVerbs(compilation);
+            }
+
+            /// <summary>
+            /// Gets relevant info from the attributes on the MVC controller action we're examining.
+            /// </summary>
+            /// <param name="attributeDatas">Attributes on the MVC controller action.</param>
+            /// <param name="verbs">Information on which HTTP verbs are specified.</param>
+            /// <param name="antiforgeryTokenDefined">Indicates that the ValidateAntiforgeryToken attribute was specified.</param>
+            /// <param name="isAction">Indicates that the MVC controller method doesn't have an attribute saying it's not really an action.</param>
+            public void ComputeAttributeInfo(
+                ImmutableArray<AttributeData> attributeDatas, 
+                out MvcHttpVerbs verbs,
+                out bool antiforgeryTokenDefined,
+                out bool isAction)
+            {
+                verbs = default(MvcHttpVerbs);
+                antiforgeryTokenDefined = false;
+                isAction = true;    // Presumed an MVC controller action until proven otherwise.
+
+                foreach (AttributeData a in attributeDatas)
+                {
+                    if (IsAttributeClass(a, this.ValidateAntiforgeryTokenAttributeSymbol))
+                    {
+                        antiforgeryTokenDefined = true;
+                    }
+                    else if (IsAttributeClass(a, this.HttpGetAttributeSymbol))
+                    {
+                        verbs |= MvcHttpVerbs.Get;
+                    }
+                    else if (IsAttributeClass(a, this.HttpPostAttributeSymbol))
+                    {
+                        verbs |= MvcHttpVerbs.Post;
+                    }
+                    else if (IsAttributeClass(a, this.HttpPutAttributeSymbol))
+                    {
+                        verbs |= MvcHttpVerbs.Put;
+                    }
+                    else if (IsAttributeClass(a, this.HttpDeleteAttributeSymbol))
+                    {
+                        verbs |= MvcHttpVerbs.Delete;
+                    }
+                    else if (IsAttributeClass(a, this.HttpPatchAttributeSymbol))
+                    {
+                        verbs |= MvcHttpVerbs.Patch;
+                    }
+                    else if (IsAttributeClass(a, this.AcceptVerbsAttributeSymbol))
+                    {
+                        if (a.AttributeConstructor.Parameters != null
```

Comment:
Redundant null check. We return an empty list for no parameters case: http://source.roslyn.io/#Microsoft.CodeAnalysis/Symbols/IMethodSymbol.cs,107